### PR TITLE
Integrate reward points

### PR DESCRIPTION
**Pull request by phattran-gft**
Integrate Reward Points into Trading template after Settle phase

### CODE REVIEW

File: src/daml/Daml/Trading/Service/Trade.daml [hunk 1]
```diff
@@ -218,5 +219,14 @@ template TradeSettlement
...
           newOwnerAccount = adminCashAccKey 
+
+        tradeSuccessfullySignal <- create TradingPoint.TradeSuccess with
+          admin = custodian
+          seller = seller
+          buyer = buyer
+          tradeAmount = amount
+          asset = show (batchCid)
+          
+        -- exercise tradeSuccessfullySignal TradingPoint.IssueRewardPoint
```
Comment (trungho2) [inline src/daml/Daml/Trading/Service/Trade.daml]: remove

File: src/daml/Daml/Triggers/TradingSuccessTrigger.daml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import Daml.Trading.Reward.TradingPoint
+import DA.Foldable (forA_)
+
+
+issueTradingPointTrigger: Trigger ()
+issueTradingPointTrigger = Trigger with
+  initialize = pure ()
+  updateState = \_ -> pure ()
+  rule = \p -> do
+    debug ("triggered")
```
Comment (trungho2) [inline src/daml/Daml/Triggers/TradingSuccessTrigger.daml]: change message

File: src/daml/Daml/Triggers/TradingSuccessTrigger.daml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    tradeData <- query @TradeSuccess
+    forA_ tradeData (\tradeSuccessdata -> do
```
Comment (trungho2) [inline src/daml/Daml/Triggers/TradingSuccessTrigger.daml]: filter tradeSuccess that has admin ==  p first

File: src/daml/Daml/Triggers/TradingSuccessTrigger.daml (hunk 2 above)
Comment (phattran-gft) [reply src/daml/Daml/Triggers/TradingSuccessTrigger.daml]: Done

File: src/daml/Daml/Triggers/TradingSuccessTrigger.daml (hunk 3 above)
Comment (phattran-gft) [reply src/daml/Daml/Triggers/TradingSuccessTrigger.daml]: Done

File: src/daml/Daml/Trading/Service/Trade.daml (hunk 1 above)
Comment (phattran-gft) [reply src/daml/Daml/Trading/Service/Trade.daml]: Done

**Review by trungho2 — APPROVED** [review]
LGTM

**Pull request merged**